registry/storage/driver/gdrive/gdrive.go

```diff
@@ -0,0 +1,488 @@
+package gdrive
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	"github.com/docker/distribution/registry/storage/driver/base"
+	"github.com/docker/distribution/registry/storage/driver/factory"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+	"golang.org/x/oauth2/jwt"
+	drive "google.golang.org/api/drive/v3"
+)
+
+const driverName = "gdrive"
+
+// DriverParameters represents all configuration options available for the
+// filesystem driver
+type DriverParameters struct {
+	KeyFile       string
+	RootDirectory string
+}
+
+func init() {
+	factory.Register(driverName, &gdriveDriverFactory{})
+}
+
+// gdriveDriverFactory implements the factory.StorageDriverFactory interface
+type gdriveDriverFactory struct{}
+
+func (factory *gdriveDriverFactory) Create(parameters map[string]interface{}) (storagedriver.StorageDriver, error) {
+	return FromParameters(parameters)
+}
+
+type driver struct {
+	rootID       string
+	driveService *drive.Service
+}
+
+type baseEmbed struct {
+	base.Base
+}
+
+// Driver is a storagedriver.StorageDriver implementation backed by Google Drive.
+type Driver struct {
+	baseEmbed
+}
+
+// FromParameters constructs a new Driver with a given parameters map
+func FromParameters(parameters map[string]interface{}) (*Driver, error) {
+	params, err := fromParametersImpl(parameters)
+	if err != nil || params == nil {
+		return nil, err
+	}
+	return New(*params), nil
+}
+
+func fromParametersImpl(parameters map[string]interface{}) (*DriverParameters, error) {
+	var (
+		keyFile       = ""
+		rootDirectory = ""
+	)
+
+	if parameters != nil {
+		if keyFileName, ok := parameters["keyFile"]; ok {
+			keyFile = fmt.Sprint(keyFileName)
+		}
+
+		if rootDirectoryName, ok := parameters["rootDirectory"]; ok {
+			rootDirectory = fmt.Sprint(rootDirectoryName)
+		}
+	}
+
+	params := &DriverParameters{
+		KeyFile:       keyFile,
+		RootDirectory: rootDirectory,
```

Comment:
`RootDirectory` should have a default value set as a const, if no default is appropriate an error should be raised.